File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_phone.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_phonetype_enum as shared_codat_datacontracts_datasets_phonetype_enum
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsPhone:
    type: shared_codat_datacontracts_datasets_phonetype_enum.CodatDataContractsDatasetsPhoneTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    number: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('number'), 'exclude': lambda f: f is None }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsphone.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsPhoneTypeEnum } from "./codatdatacontractsdatasetsphonetypeenum";
import { Expose } from "class-transformer";


export class CodatDataContractsDatasetsPhone extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "number" })
  number?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type: CodatDataContractsDatasetsPhoneTypeEnum;
}